cpp/src/io/json/json_gpu.cu

```diff
@@ -281,26 +280,26 @@ struct ConvertFunctor {
    *
    * It is handled here rather than within convertStrToValue() as that function
    * is used by other types (ex. timestamp) that aren't 'booleable'.
-   **/
+   */
   template <typename T, typename std::enable_if_t<std::is_integral<T>::value> * = nullptr>
-  __host__ __device__ __forceinline__ bool operator()(const char *data,
+  __host__ __device__ __forceinline__ bool operator()(char const *begin,
+                                                      char const *end,
                                                       void *output_columns,
-                                                      long row,
-                                                      long start,
-                                                      long end,
+                                                      cudf::size_type row,
                                                       const ParseOptions &opts)
   {
     T &value{static_cast<T *>(output_columns)[row]};
 
     // Check for user-specified true/false values first, where the output is
     // replaced with 1/0 respectively
-    const size_t field_len = end - start + 1;
-    if (serializedTrieContains(opts.trueValuesTrie, data + start, field_len)) {
+    const size_t field_len = end - begin;
+    if (serializedTrieContains(opts.trueValuesTrie, begin, field_len)) {
       value = 1;
-    } else if (serializedTrieContains(opts.falseValuesTrie, data + start, field_len)) {
+    } else if (serializedTrieContains(opts.falseValuesTrie, begin, field_len)) {
       value = 0;
     } else {
-      value = decode_value<T>(data, start, end, opts);
+      // TODO: refactor decode_value to use pointers too
+      value = decode_value<T>(begin, 0, field_len - 1, opts);
```

Comment:
This can be replaced with I+LE:
```cpp
    value = [&, field_len = end - begin]() -> T {
      if (serializedTrieContains(opts.trueValuesTrie, begin, field_len)) {
        return 1;
      } else if (serializedTrieContains(opts.falseValuesTrie, begin, field_len)) {
        return 0;
      } else {
        // TODO: refactor decode_value to use pointers too
        return decode_value<T>(begin, 0, field_len - 1, opts);
      }
    }();
```
Also, why is it necessray to pass `void* output_columns` and that immediately cast to `T *`? Can't we just pass `T *`, which would enable:
```
output_columns[row] = [] () ...
```
Also also, seems odd we are always returning `true` from this function - is that necessary?

Reply:
We have to pass void* because we don't have the type at the caller site - we have a list of dtype instead. The parameter is poorly named, fixed that.
The operator returns whether the decoded value is NaN. For some types the answer is always no, but we still have to return it.

Reply:
I see, that makes sense! Thanks!